### PR TITLE
Snapshot resource (PowerScale 1.1)

### PR DESCRIPTION
**Pull request by doriac11**
# Description
Create Update Delete Snapshot Resource 

# ISSUE TYPE
- Feature Pull Request

##### RESOURCE OR DATASOURCE NAME
Snapshot Resource.

##### OUTPUT
image

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [x] I have maintained at least 80% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have maintained backward compatibility

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [x] Unit tests
- [x] Acceptance tests

### CODE REVIEW

**Comment by doriac11** [comment]
run e2e test

**Comment by doriac11** [comment]
run e2e test

**Review by shekhar-j — APPROVED** [review]

**Comment by doriac11** [comment]
run e2e test

**Comment by doriac11** [comment]
run e2e test

File: powerscale/provider/snapshot_resource.go [hunk 1]
```diff
@@ -0,0 +1,326 @@
...
+			"Path variable is not able to be updated after creation",
+		)
+		return
+	}
+
+	// Populate the Edit Parameters
+	editVal := powerscale.V1SnapshotSnapshotExtendedExtended{}
+	if plan.Name != state.Name {
+		editVal.Name = plan.Name.ValueStringPointer()
+		state.Name = plan.Name
```
Comment (nikitajoshi1) [inline powerscale/provider/snapshot_resource.go:278]: @doriac11 Why are we updating the state with plan data even before modify request is sent? Ideally state should have been updated only once modify is successful.

File: powerscale/helper/snapshot_helper.go [hunk 2]
```diff
@@ -32,14 +34,88 @@ func GetAllSnapshots(ctx context.Context, client *client.Client) ([]powerscale.V
 	return result.GetSnapshots(), err
 }
 
+// GetSpecificSnapshot returns the full list of snapshots.
```
[the remark this answers is not in the document]
Comment (doriac11) [reply powerscale/helper/snapshot_helper.go:37]: Updated

File: powerscale/provider/snapshot_resource.go (hunk 1 above)
Comment (doriac11) [reply powerscale/provider/snapshot_resource.go:278]: Me and @nikitajoshi1 discussed in slack, this solution should be suitable.

**Review by nikitajoshi1 — APPROVED** [review]

**Pull request merged**